### PR TITLE
Remove availability and fallbacks for iOS 11 and macOS 10.13

### PR DESCRIPTION
**Pull request by jmagman**
macOS minimum is 10.13: https://github.com/flutter/buildroot/pull/588
iOS minimum is 11: https://github.com/flutter/buildroot/pull/574

- Remove `@available` checks for lower versions.  
- The engine builders are running Xcode 13 so remove outdated `*_VERSION_MAX_ALLOWED` checks.
- Remove `NS_AVAILABLE_IOS` annotations in implementation files.

Part of https://github.com/flutter/flutter/issues/101959
Part of https://github.com/flutter/flutter/issues/107741

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: shell/platform/darwin/ios/framework/Source/FlutterPlatformPlugin.mm [hunk 1]
```diff
@@ -17,7 +17,7 @@
 constexpr char kTextPlainFormat[] = "text/plain";
 const UInt32 kKeyPressClickSoundId = 1306;
 
-}  // namespaces
+}  // namespace
```
Comment (jmagman) [inline shell/platform/darwin/ios/framework/Source/FlutterPlatformPlugin.mm:20]: formatter wanted to touch this.

File: shell/platform/darwin/ios/framework/Source/VsyncWaiterIosTest.mm [hunk 2]
```diff
@@ -17,7 +17,7 @@
   auto runner = thread->GetTaskRunner();
   return runner;
 }
-}
+}  // namespace
```
Comment (jmagman) [inline shell/platform/darwin/ios/framework/Source/VsyncWaiterIosTest.mm:20]: Formatter wanted this too.

**Review by chinmaygarde — APPROVED** [review]
I know I am being greedy but here goes. What are the consequences of updated to iOS 12.0 and macOS 14? MTLEvent and friends are available since then and assuming their availability would make a fallback path using host synchronization unnecessary. I've been budgeting time to add that fallback but this would be a nice win. Not sure high priority though.

**Comment by jmagman** [comment]
> What are the consequences of updated to iOS 12.0 and macOS 14?

I'm usually more concerned about dropping hardware than OS versions since it's "free" to upgrade, but expensive to buy new hardware.  Fortunately iOS 12 (2018) didn't drop any devices supported by iOS 11.  iOS 11 isn't even measured in this dataset https://gs.statcounter.com/os-version-market-share/ios/mobile-tablet/worldwide (for what that's worth).  

macOS 10.14 Mojave also came out 2018.  Mojave requires a Metal capable chip, so we would be able to drop the GL backend https://github.com/flutter/flutter/issues/108304 (https://support.apple.com/en-us/HT208898).  According to https://gs.statcounter.com/os-version-market-share/macos/desktop/worldwide macOS 10.13 High Sierra has about 4.5% market share.  Probably more importantly, hardware manufactured 2009-2012/5 was dropped, and some users hang on to their Macs as long as possible.

10.13 High Sierra
```
MacBook (Late 2009 or newer)
MacBook Pro (Mid 2010 or newer)
MacBook Air (Late 2010 or newer)
Mac mini (Mid 2010 or newer)
iMac (Late 2009 or newer)
Mac Pro (Mid 2010 or newer)
```
https://support.apple.com/kb/sp765?locale=en_US

vs 10.14 Mojave
```
MacBook (Early 2015 or newer)
MacBook Air (Mid 2012 or newer)
MacBook Pro (Mid 2012 or newer)
Mac mini (Late 2012 or newer)
iMac (Late 2012 or newer)
iMac Pro (2017)
Mac Pro (Late 2013; Mid 2010 and Mid 2012 models with [recommended Metal-capable graphics cards](https://support.apple.com/en-us/HT208898))
```
https://support.apple.com/kb/SP777?locale=en_US

Small overall end-user market share doesn't necessarily mean we should drop it if it also means also dropping a disproportionate number of NBU customers, and supporting "older platforms is one of our differentiators".
So I'd always err on the side of avoiding dropping old versions until we absolutely have to, or the maintenance burden becomes too great, or other reasons cited in go/flutter-deprecating-old-platforms.

**Pull request merged**